src/griffin/test/contract.cljc

```diff
@@ -204,8 +206,10 @@
                       (cleanup impl calls)))})))
 
 (defn gen-valid-args [state method]
-  (gen/such-that (fn [args] (p/precondition method state args))
-                 (p/args method state)))
+  (let [gen-args (p/args method state)]
+    (validate! gen/generator? gen-args (str ":args must return a generator for " method))
```

Comment:
Commentary: It might be nicer to merge extra context into `validate!`, to expose the full `method` and `state` in the error.